modules/compress/src/main/java/org/apache/ignite/internal/processors/compress/CompressionProcessorImpl.java

```diff
@@ -170,6 +177,28 @@ private ByteBuffer doCompactPage(ByteBuffer page, int pageSize) throws IgniteChe
         return compactPage;
     }
 
+    /** Check if filesystem actually supports punching holes. */
+    private void checkPunchHole(Path storagePath, int fsBlockSz) throws IgniteException {
+        Path testFile = storagePath.resolve("punch_hole_" + UUID.randomUUID() + ".test");
```

Comment:
Why not `File createTempFile`?

Reply:
Because it will be created in /tmp. We should check exactly the given path

Reply:
`public static File createTempFile(String prefix, String suffix, File directory)` can be used